### ISSUE TITLE
False negative: Wrong number of arguments for `isinstance`

### ISSUE THREAD

**Issue by nickdrozd** [pylint-dev/pylint#9847]
### Bug description

```python
isinstance(5)
isinstance(5, int, int)
```

### Command used

```shell
pylint
```

### Pylint output

```python
No warnings
```

### Expected behavior

Pylint should warn that the number of arguments is wrong.

I don't know if this affects other builtin functions.

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.11.1 (main, Dec  7 2022, 01:11:44) [GCC 9.4.0]
```

**Comment by Pierre-Sassoulas** [comment]
Feel like something that is more suited for a type checker like mypy. (Similar to https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/invalid-envvar-default.html that we probably wouldn't do today.)

**Comment by rogersheu** [comment]
It appears that https://github.com/pylint-dev/pylint/blob/466c671cbe5c4f23747b8684665cebefdf6d8fdf/pylint/checkers/typecheck.py#L1455 but perhaps this makes it a potential `TODO`?

**Comment by nickdrozd** [comment]
@rogersheu Fixing it for all builtins might be difficult, but fixing it just for `isinstance` looks easy:

```python
    def _check_isinstance_args(self, node: nodes.Call) -> None:
        if len(node.args) != 2:
            # isinstance called with wrong number of args
            return
```
Instead of returning, raise the appropriate error instead.

**Comment by rogersheu** [comment]
Thanks for pointing that out! I've opened a PR to address this 👌 .
